Update Ocean to blaze 3.8

geoffrey4444 opened the pull request:
## Proposed changes

Update Blaze on ocean to v3.8.

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
This PR just switches to use blaze 3.8 on the cluster Ocean at Cal State Fullerton, which is needed after spectre switched to requiring blaze 3.8.

### Code review checklist

- [x] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [x] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [x] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

geoffrey4444 committed:
Update Ocean to blaze 3.8

fmahebert commented:
[not a change request] What about the other Ocean environments? Are they still supported?

kidder commented:
@geoffrey4444 this needs a rebase

geoffrey4444 commented:
Thanks for taking the quick looks, @kidder and @fmahebert ! I rebased on latest develop and made the following changes:

1) I continue to support ocean_clang.sh and ocean_gcc.sh. Both use the same modules but the clang one uses clang instead of gcc when compiling spectre. Both of these are suitable for running jobs on ocean's orca-1 queue (the newer, faster queue).
2) I dropped support for ocean_orca0_gcc.sh. This let us run on our older compute nodes but required compiling on those older, slower compute nodes, which was a big pain. Once we're doing a lot of production runs with spectre, I might bring this support back with updated modules, but until then, I decided it's not worth the time, since no one was actually running spectre on the orca0 queue as long as I can remember.

kidder commented on support/Environments/ocean_gcc.sh:
```diff
@@ -18,24 +18,25 @@ spectre_setup_modules() {
     module unload openblas-0.3.4-gcc-7.3.0-tt2coe7
     module unload python/3.7.0
-    module unload charm-6.10.2-libs
+    module unload charm-6.10.2-libss
```
fix this

kidder reviewed (changes requested):
you can squash
